core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/GoogleProjectRegistrationComponent.scala

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.model.{
+  GoogleProjectId,
+  GoogleProjectRegistration,
+  RawlsBillingAccountName,
+  RawlsBillingProjectName
+}
+
+final case class GoogleProjectRegistrationRecord(googleProjectId: String,
+                                                 billingAccount: Option[String],
+                                                 message: Option[String],
+                                                 billingProjectId: String
+)
+
+object GoogleProjectRegistrationRecord {
+  def fromGoogleProjectRegistration(googleProjectReg: GoogleProjectRegistration): GoogleProjectRegistrationRecord =
+    GoogleProjectRegistrationRecord(
+      googleProjectReg.googleProjectId.value,
+      googleProjectReg.billingAccount.map(_.value),
+      googleProjectReg.message,
+      googleProjectReg.billingProjectId.value
+    )
+
+  def toGoogleProjectRegistration(googleProjectRegRec: GoogleProjectRegistrationRecord): GoogleProjectRegistration =
+    GoogleProjectRegistration(
+      GoogleProjectId(googleProjectRegRec.googleProjectId),
+      googleProjectRegRec.billingAccount.map(RawlsBillingAccountName(_)),
+      googleProjectRegRec.message,
+      RawlsBillingProjectName(googleProjectRegRec.billingProjectId)
+    )
+
+}
+
+trait GoogleProjectRegistrationComponent {
+  this: DriverComponent =>
+
+  import driver.api._
+
+  class GoogleProjectRegistrationTable(tag: Tag) extends Table[GoogleProjectRegistrationRecord](tag, "GOOGLE_PROJECT") {
+    def googleProjectId = column[String]("GOOGLE_PROJECT_ID", O.PrimaryKey, O.Length(254))
+
+    def billingAccount = column[Option[String]]("BILLING_ACCOUNT")
+
+    def message = column[Option[String]]("MESSAGE")
+
+    def billingProject = column[String]("BILLING_PROJECT")
+
+    def * = (googleProjectId, billingAccount, message, billingProject) <> (
+      (GoogleProjectRegistrationRecord.apply _).tupled,
+      GoogleProjectRegistrationRecord.unapply
+    )
+  }
+
+  type GoogleProjectRegistrationQuery = Query[GoogleProjectRegistrationTable, GoogleProjectRegistrationRecord, Seq]
+
+  object googleProjectRegistrationQuery extends TableQuery(new GoogleProjectRegistrationTable(_)) {
+
+    def create(googleProjectReg: GoogleProjectRegistration): ReadWriteAction[GoogleProjectRegistration] =
+      googleProjectRegistrationQuery
+        .withId(googleProjectReg.googleProjectId.value)
+        .result
+        .flatMap {
+          case Seq() =>
+            googleProjectRegistrationQuery += GoogleProjectRegistrationRecord.fromGoogleProjectRegistration(
+              googleProjectReg
+            )
+          case _ =>
+            updateBillingProjectIdAndAccount(googleProjectReg)
```

Comment:
This is my first pass at just making sure updating works, but I think the service layer should probably either call create or update depending on what it finds instead of having the db layer automatically update on create.  But then the question is, what should this layer do if the google project already exists?

Reply:
In theory, we should skip this layer if we've found that the project already exists. 

However, is there a case where the DB has successfully recorded the google project but assigning the billing failed? I'm not sure how rawls handles partially successful operations.

Reply:
Ah, I thought if the project already existed but the request had a different billing project, then we should update the billing project.  Or is that supposed to be a different API altogether, and for this one we never update it?

Reply:
From a feature standpoint, I don't _think_ rawls supports change the terra billing project for a workspace. (and we also don't allow updating the TDR billing profile associated with resources and therefore their google project). 
 However, we do allow updating the underlying google account, but that is a separate endpoint. 
 
 From a "let's make this operation idempotent" - Idk. There may be a case for just overwriting the existing billing project, but that might be a case where we should throw an error. I'm not sure.

Reply:
@marctalbott and/or @pshapiro4broad can you weigh in on the appropriate behavior here?  If this API is called with a google project ID that is already registered in the rawls database, should we:
a) do nothing
b) throw an error
c) check if the requested billing project id matches; if it does, do nothing, if it doesn't:
   c.1) throw an error
   c.2) update the billing project id
Any other edge cases I'm forgetting?